components/Books.js

```diff
@@ -53,9 +53,14 @@ export default class Books extends React.Component {
       loader: true
     });
 
+    let email = "";
```

Comment:
Logically the way you have handled email is absolutely correct but it seems there's slight compromise with the readability, what if you check for email and build the url inside the cases itself?